docs/data/date-pickers/custom-field/custom-behavior/ReadOnlyMaterialTextField.tsx

```diff
@@ -1,47 +1,65 @@
 import * as React from 'react';
-import dayjs, { Dayjs } from 'dayjs';
+import { Dayjs } from 'dayjs';
 import TextField from '@mui/material/TextField';
-import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
 import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
 import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
-import { DatePicker, DatePickerFieldProps } from '@mui/x-date-pickers/DatePicker';
+import {
+  DatePicker,
+  DatePickerProps,
+  DatePickerFieldProps,
+} from '@mui/x-date-pickers/DatePicker';
 import { useValidation, validateDate } from '@mui/x-date-pickers/validation';
-import { useSplitFieldProps } from '@mui/x-date-pickers/hooks';
+import {
+  useSplitFieldProps,
+  useParsedFormat,
+  usePickersContext,
+} from '@mui/x-date-pickers/hooks';
 
-function ReadOnlyField(props: DatePickerFieldProps<Dayjs, false>) {
+function ReadOnlyDateField(props: DatePickerFieldProps<Dayjs, false>) {
   const { internalProps, forwardedProps } = useSplitFieldProps(props, 'date');
-
   const { value, timezone, format } = internalProps;
   const { InputProps, slotProps, slots, ...other } = forwardedProps;
 
+  const pickersContext = usePickersContext();
+
+  const parsedFormat = useParsedFormat(internalProps);
   const { hasValidationError } = useValidation({
     validator: validateDate,
     value,
     timezone,
     props: internalProps,
   });
 
+  const handleTogglePicker = (event: React.UIEvent) => {
+    if (pickersContext.open) {
+      pickersContext.onClose(event);
+    } else {
+      pickersContext.onOpen(event);
+    }
+  };
+
   return (
     <TextField
       {...other}
       value={value == null ? '' : value.format(format)}
+      placeholder={parsedFormat}
       InputProps={{ ...InputProps, readOnly: true }}
```

Comment:
Offtopic nitpick: it might make sense to add a `pointer: 'cursor'` style to the `TextField` or at least `inputProps` to reflect the behavior better. 👍

Reply:
Done :+1: